src/core/fbcontainer.h

```diff
@@ -155,6 +167,46 @@ namespace forte {
         TFBContainerList mChildren; //!< List of children (i.e, fbs or subapplications in this container)
     };
 
+    template<typename T>
+    class CInternalFB {
+        static_assert(std::is_base_of_v<CFunctionBlock, T>, "T must be a CFunctionBlock");
+      public:
+        CInternalFB(CStringDictionary::TStringId paInstanceNameId, CFBContainer &paContainer) :
+                mFB(std::make_unique<T>(paInstanceNameId, paContainer)) {
+          paContainer.addFB(*mFB);
+        }
+
+        CInternalFB(CStringDictionary::TStringId paInstanceNameId, const char *paConfigString,
+                    CFBContainer &paContainer) :
+                mFB(std::make_unique<T>(paInstanceNameId, paContainer)) {
+          mFB->configureFB(paConfigString);
+          paContainer.addFB(*mFB);
+        }
+
+        ~CInternalFB() {
+          mFB->getParent().removeFB(*mFB);
+        }
+
+        T *get() noexcept {
+          return mFB.get();
+        }
+
+        T &operator*() noexcept {
+          return mFB.operator*();
+        }
+
+        T *operator->() noexcept {
```

Comment:
same

Reply:
We need the `->` operator for access to members of the wrapped FB class (e.g., `fb_FB1->var_DI1`).

Reply:
ah forgot about that.